### PR TITLE
Ford: hybrids are unsupported

### PR DESCRIPTION
**Pull request by sshane**
Dashcamming these cars for the release, will revisit the PCM checksum bug after. From 2k rlogs, that issue only happens on HEV and PHEVs, will check qlogs soon!

These bits are always the same value across segments and are the same as on other hybrid or ICE cars, and they mismatch between hybrid and ICE.

I didn't exclude first or last segments, so that means this signal is likely continually sent and valid before and after the car is started, ~~will verify~~ it is.

```python
addr: 0x230, bidx: (7, 1) - hybrid: {1}, ice: {0}  # TransGearData, signal not in DBC
addr: 0x230, bidx: (7, 0) - hybrid: {1}, ice: {0}  # TransGearData, signal not in DBC
addr: 0x43b, bidx: (0, 2) - hybrid: {1}, ice: {0}  # not in DBC
addr: 0x5c , bidx: (0, 1) - hybrid: {1}, ice: {0}  # Gear_Shift_by_Wire_3->TrnSbwSysHlth_D_Actl
addr: 0x5c , bidx: (5, 2) - hybrid: {0}, ice: {1}  # Gear_Shift_by_Wire_3->TrnLvrV_D_Rq
addr: 0x25c, bidx: (0, 3) - hybrid: {1}, ice: {0}  # not in DBC
addr: 0x25c, bidx: (0, 6) - hybrid: {1}, ice: {0}  # not in DBC
addr: 0x167, bidx: (5, 4) - hybrid: {0}, ice: {1}  # VehicleOperatingModes->TrnAinTq_D_Qf
addr: 0x167, bidx: (5, 3) - hybrid: {0}, ice: {1}  # VehicleOperatingModes->TrnAinTq_D_Qf
```

### CODE REVIEW

File: selfdrive/car/ford/interface.py [hunk 1]
```diff
@@ -78,6 +78,8 @@ def _update(self, c):
     events = self.create_common_events(ret, extra_gears=[GearShifter.manumatic])
     if not self.CS.vehicle_sensors_valid:
       events.add(car.CarEvent.EventName.vehicleSensorsInvalid)
+    if self.CS.hybrid_platform:
+      events.add(car.CarEvent.EventName.dashcamMode)
```
Comment (sshane) [inline selfdrive/car/ford/interface.py:82]:
```suggestion
      events.add(car.CarEvent.EventName.startupNoControl)
```

File: selfdrive/car/ford/carstate.py [hunk 2]
```diff
@@ -17,10 +17,16 @@ def __init__(self, CP):
...
     self.vehicle_sensors_valid = False
+    self.hybrid_platform = False
 
   def update(self, cp, cp_cam):
     ret = car.CarState.new_message()
 
+    # Hybrid variants experience a bug where a message from the PCM sends invalid checksums,
+    # we do not support these cars at this time.
+    # TrnAin_Tq_Actl and its quality flag are only set on ICE platform variants
+    self.hybrid_platform = cp.vl["VehicleOperatingModes"]["TrnAinTq_D_Qf"] == 0
```
Comment (sshane) [inline selfdrive/car/ford/carstate.py:28]:
```suggestion
    self.hybrid_platform = cp.vl["VehicleOperatingModes"]["TrnAinTq_D_Qf"] == 0
```

**Comment by sshane** [comment]
This signal exists at 100Hz on all segments checked, it's always 0 when hybrid, and always 3 when ICE.

I tested letting the ECU go to sleep on our Bronco Sport (waited until even the ABS module stopped sending), and pressing brake or pressing push-to-start button without brake, and the first message it sent when it came back up was the Qf at 3--meaning there's no initialization, it's hard-coded to 3.

**Comment by incognitojam** [comment]
related #28124

**Comment by paulmg377** [comment]
Anything we can do to help get support for hybrids? 2021- ford escape hybrid- with 3x and q3 harness.

**Pull request merged**